### PR TITLE
modules: nrfxlib: nrf_802154: adjust ipct channels configuration.

### PR DESCRIPTION
**Pull request by piotrkoziar**
1. Adjusts ipct channels to require one allocated sink-source pair by using the same channel for sink and source.

2. Adds compile-time checks that will assert if the device tree has missing dppi or ipct connections.

depends on https://github.com/nrfconnect/sdk-zephyr/pull/1913

test-fw-nrfconnect-rs: PR-1231

### CODE REVIEW

File: modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_grtc_hw_task.c [hunk 1]
```diff
@@ -70,6 +70,92 @@
...
+	     "The required DPPIC_G2 channel is not reserved");
+
+/* Ensure something similar to this is present in the DT:
+ * &cpurad_ipct {
+ *	sink-channel-links = <2 13 2>;
+ * };
+ * where first 2 corresponds to IPCT_L_HT_CHANNEL.
+ */
+#define _HT_DT_CHECK_IPCT_L_LINK(node_id, prop, idx) \
+	((DT_PROP_BY_IDX(node_id, prop, idx) == IPCT_L_HT_CHANNEL) && (idx%3 == 0)) ||
```
Comment (ankuns) [inline modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_grtc_hw_task.c:121]:
```suggestion
	((DT_PROP_BY_IDX(node_id, prop, idx) == IPCT_L_HT_CHANNEL) && ((idx) % 3 == 0)) ||
```

File: modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_grtc_hw_task.c [hunk 2]
```diff
@@ -70,6 +70,92 @@
...
+	     "The required IPCT_L link is not reserved");
+
+/* Ensure something similar to this is present in the DT:
+ * &ipct130 {
+ *	source-channel-links = <2 3 2>;
+ * };
+ * where first 2 corresponds to IPCT_G1_HT_CHANNEL.
+ */
+#define _HT_DT_CHECK_IPCT_G1_LINK(node_id, prop, idx) \
+	((DT_PROP_BY_IDX(node_id, prop, idx) == IPCT_G1_HT_CHANNEL) && (idx%3 == 0)) ||
```
Comment (ankuns) [inline modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_grtc_hw_task.c:143]:
```suggestion
	((DT_PROP_BY_IDX(node_id, prop, idx) == IPCT_G1_HT_CHANNEL) && ((idx) % 3 == 0)) ||
```

File: modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_timestamper.c [hunk 3]
```diff
@@ -75,6 +75,93 @@ static int32_t m_timestamp_cc_channel;
...
+	     "The required DPPIC_G2 channel is not reserved");
+
+/* Ensure something similar to this is present in the DT:
+ * &cpurad_ipct {
+ *	source-channel-links = <2 13 2>;
+ * };
+ * where first 2 corresponds to IPCT_L_TS_CHANNEL.
+ */
+#define _TS_DT_CHECK_IPCT_L_LINK(node_id, prop, idx) \
+	((DT_PROP_BY_IDX(node_id, prop, idx) == IPCT_L_TS_CHANNEL) && (idx%3 == 0)) ||
```
Comment (ankuns) [inline modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_timestamper.c:125]:
```suggestion
	((DT_PROP_BY_IDX(node_id, prop, idx) == IPCT_L_TS_CHANNEL) && ((idx) % 3 == 0)) ||
```

File: modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_timestamper.c [hunk 4]
```diff
@@ -75,6 +75,93 @@ static int32_t m_timestamp_cc_channel;
...
+
+BUILD_ASSERT(TS_DT_HAS_RESERVED_DPPIC_G1_CHANNEL,
+	     "The required DPPIC_G1 channel is not reserved");
+
+/* Ensure something similar to this is present in the DT:
+ * &dppic132 {
+ *	owned-channels = <3>;
+ *	sink-channels = <3>;
+ * where 3 corresponds to DPPIC_G2_TS_CHANNEL.
+ * };
```
Comment (ankuns) [inline modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_timestamper.c:103]: To preserve consistency
```suggestion
 * };
 * where 3 corresponds to DPPIC_G2_TS_CHANNEL.
```

File: west.yml [hunk 5]
```diff
@@ -63,7 +63,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 4c996ecd07e576524a98405b2af6c01c56cb83e5
+      revision: a7b0003fac963c2e9f9f0edbc9fabc552d7981b4
```
Comment (ankuns) [inline west.yml:66]: While https://github.com/nrfconnect/sdk-zephyr/pull/1913 is sill not merged
`pull/1913/head` is used (unless something changed in recent weeks?)

**Comment by dawidprzybylo** [comment]
RS CI plan is expected to fail on Integration because PR-1231 on RS TAS should also be merged after this PR, however
RS are green(https://jenkins-ncs.nordicsemi.no/job/latest/job/test-fw-nrfconnect-rs_drv154_integration/job/PR-1231/3/) 
it is not a blocker here

**Comment by piotrkoziar** [comment]
As @dawidprzybylo mentioned, after merging this PR please merge [**PR to test-fw-nrfconnect-rs**](https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-rs/pull-requests/1231/overview) immediately (or ping @piotrkoziar @dawidprzybylo @ankuns so we can do this)

**Review by ankuns — APPROVED** [review]

**Review by maciejbaczmanski — APPROVED** [review]

**Review by ArekBalysNordic — APPROVED** [review]

**Pull request merged**